metpy/calc/indices.py

```diff
@@ -51,3 +51,49 @@ def precipitable_water(dewpt, pressure, top=400 * units('hPa')):
     pw = -1. * (np.trapz(w.magnitude, pres_layer.magnitude) * (w.units * pres_layer.units) /
                 (g * rho_l))
     return pw.to('millimeters')
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]')
+def mean_wind_pressure_weighted(u, v, p, heights=None, bottom=None, depth=None):
+    r"""Calculate pressure-weighted mean wind through a layer.
+
+    Layer top and bottom specified in height or pressure.
+
+    Parameters
+    ----------
+    u : `pint.Quantity`
+        U-component of wind.
+    v : `pint.Quantity`
+        V-component of wind.
+    p : `pint.Quantity`
+        Atmospheric pressure profile
+    heights : `pint.Quantity`, optional
+        Heights from sounding
+    bottom: `pint.Quantity`, optional
+        The bottom of the layer in either the provided height coordinate
+        or in pressure. Don't provide in meters AGL unless the provided
+        height coordinate is meters AGL. Default is the first observation,
+        assumed to be the surface.
+    depth: `pint.Quantity`, optional
+        The depth of the layer in meters or hPa.
+
+    Returns
+    -------
+    `pint.Quantity`
+        u_mean: u-component of layer mean wind.
+    `pint.Quantity`
+        v_mean: v-component of layer mean wind.
+
+    """
+    u = u.to('meters/second')
```

Comment:
Why are we doing any unit conversion at all? This calculation looks unit-agnostic.